### ISSUE TITLE
add pre-commit/pre-commit-ci to type-checking/ci pipeline

### ISSUE THREAD

**Issue by SiQube** [aeye-lab/pymovements#67]
## Description of the problem
- I'm always frustated when I try create a PR and e.g. flake8 fails, then I have to check in the ci protocol what went wrong, fix it and repush.
- adding [pre-commit](https://www.pre-commit.com) would enable me to check whether flake8, mypy would indicate if the check would fail and what before creating a PR

## Description of a solution
steps of the proposed solution:
- add a [pre-commit config](https://pre-commit.com/#install)
- add [pre-commit.ci](https://pre-commit.ci/)

## Minimum acceptance criteria
- [ ] non-failing pre-commit-ci checks 

## Sample Code

pre-commit config I use in my projects:
```console
repos:
-   repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.3.0
    hooks:
    -   id: check-docstring-first
    -   id: debug-statements
    -   id: end-of-file-fixer
    -   id: trailing-whitespace
-   repo: https://github.com/asottile/reorder_python_imports
    rev: v3.9.0
    hooks:
    -   id: reorder-python-imports
-   repo: https://github.com/asottile/pyupgrade
    rev: v3.2.0
    hooks:
    -   id: pyupgrade
        args: [--py36-plus]
-   repo: https://github.com/asottile/add-trailing-comma
    rev: v2.3.0
    hooks:
    -   id: add-trailing-comma
        args: [--py36-plus]
-   repo: https://github.com/PyCQA/flake8
    rev: 5.0.4
    hooks:
    -   id: flake8
        args: ['--max-line-length=100', '--ignore=E402,W504']
-   repo: https://github.com/pre-commit/mirrors-mypy
    rev: v0.982
    hooks:
    -   id: mypy
        additional_dependencies: [types-all]
```

explanations some copied from [pre-commit-hooks](https://github.com/pre-commit/pre-commit-hooks):
`check-docstring-first`
checks for a common error of placing code before the docstring.

`debug-statements`
check for debugger imports and py37+ breakpoint() calls in python source.

`end-of-file-fixer`
checks for new line at the end of the file

`trailing-whitespace`
checks for trailing whitespaces at the end of a line

`reorder-python-imports`
reorders python imports and hence prevents merge conflicts because of imports: [reorder python imports](https://github.com/asottile/reorder_python_imports)

`pyupgrade`
checks whether the code is up to the newest python standard, e.g. f-string vs .format() [pyupgrade](https://github.com/asottile/pyupgrade)

`add-trailing-comma`
automatically adds trailing commas where needed [add trailing comma](https://github.com/asottile/add-trailing-comma)

`flake8`
checks flake8

`mypy`
checks mypy

[pre-commit-ci](https://pre-commit.ci/) enables pre-commit in ci and also auto fixes things such as trailing comma, trailing white space etc. also automatically checks that versions of all the pre-commit hooks are the newest version (autoupdate).

moreover, pre-commit-ci is super fast compared to github actions

## Context

I use this tool and it is very useful.

if you want to commit code without checking you can use `git commit --no-verify`

**Comment by dkrako** [comment]
One additional note:
when integrating `flake8` or `black` into the pre-commit, we will have to make sure that the exact same configuration as in `tox.ini` is used.
I wouldn't like to have definitions at two places, but there will be a solution for this for sure.

**Comment by SiQube** [comment]
#70 adds pre-commit -- afterwards we can discuss using [pre-commit-ci](https://pre-commit.ci), a ci that is specialized in running pre-commit and automatically fixing "easy" mistakes, e.g. trailing-whitespace. it is also out-of-the-box able to run/test flake8, black, mypy, ... moreover, pre-commit-ci only needs the config file, pushed in #70 `.pre-commit-config.yaml`, for configuration.

**Comment by dkrako** [comment]
merged in #70
